bpf-gpl/parsing.h

```diff
@@ -22,6 +22,11 @@
 #define PARSING_ERROR -1
 #define PARSING_ALLOW_WITHOUT_ENFORCING_POLICY -2
 
+#define ETHHDR(x) ((struct ethhdr *)x->data_start)
```

Comment:
nit: these only work on tc context, what if we prefix them with TC_ ? What if they were functions that return the typecasted type and take typed cali_tc_ctx input? Would it work?

Why not to define them in types.h where the cali_tc_ctx is defined?

Reply:
Just noticed that in types.h only the structures are defined, and the rest of functions are either in skb.h or parsing.h. It looks like those 3 files has a lot in common. The functions are little scattered, for example `ip_dec_ttl()` is in bpf.h. Maybe we should sort them out?

Reply:
> these only work on tc context, what if we prefix them with TC_ ? What if they were functions that return the typecasted type and take typed cali_tc_ctx input? Would it work?
That would definitely work. Honestly between macros and functions, I prefer functions:-D I'll switch to function.

Reply:
@tomastigera I think you right. it is better to move these to types.h.